# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86-cvs/xconsole-pty.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./programs/xconsole/xconsole.c.orig	Sun Mar 21 08:35:37 1999
+++ ./programs/xconsole/xconsole.c	Thu Nov 23 16:15:23 2000
@@ -786,6 +786,21 @@
 		return 0;
 	}
 #else
+#ifdef __GLIBC__	/* The new way to open a Pseudo Terminal */
+	if ((*pty = getpt()) < 0)
+		return 1;
+	strcpy(ttydev, ptsname(*pty));
+	if ((unlockpt(*pty)) < 0) {
+		close(*pty);
+		return(1);
+	}
+	if ((grantpt(*pty)) < 0) {
+		close(*pty);
+		return(1);
+	}
+	if ((*tty = open (ttydev, O_RDWR)) >= 0)
+		return(0);
+#else
 	strcpy (ttydev, "/dev/ttyxx");
 	strcpy (ptydev, "/dev/ptyxx");
 	while (PTYCHAR1[letter]) {
@@ -811,6 +826,7 @@
 	    devindex = 0;
 	    (void) letter++;
 	}
+#endif
 #endif /* sgi else not sgi */
 #endif /* CRAY else not CRAY */
 #endif /* umips && SYSTYPE_SYSV */
